### DISCUSSION TITLE
Expand Directory Size with additional drive but keep using the same directory

### DISCUSSION THREAD

**Discussion by MichalisDBA** [trapexit/mergerfs#1301] [Q&A] [answered]
Hi. 

I want to use MergeFS but i don't know if the following scenario is possible.

We have an app that writes data to `/app/data` directory.

The VM is configured without LVM. The `/app/data` is under `/` partition.

The problem is the disk (sda) is near full.

I was searching for a way to insert another disk to the VM (sdb) and somehow (with mergefs?) expand the size of sda under `/app/data` directory??

For example, this is my current df -kh command from my VM

```
+------------+------+------+-------+------+------------+
| Filesystem | Size | Used | Avail | Use% | Mounted on |
+------------+------+------+-------+------+------------+
| /dev/sda1  | 92G  | 81G  | 11G   | 88%  | /          |
+------------+------+------+-------+------+------------+
```

I want to add another 100GB drive (sdb) and somehow the above df command be like

```
+------------+------+------+-------+------+------------+
| Filesystem | Size | Used | Avail | Use% | Mounted on |
+------------+------+------+-------+------+------------+
| /dev/sda1  | 192G | 81G  | 111G  | 42%  | /          |
+------------+------+------+-------+------+------------+
```

Is this possbile with mergefs and how?

If it is not possible. Can i use the sdb disk to expand the `/app/data` directory? So that the application will not run out of space?

**Comment by trapexit** [comment]
Yes, mergerfs can combine paths. That's it's whole purpose in life... but why not create a new virtual drive and mount it to /app/data or wherever?

**Reply by MichalisDBA** [reply]
Hello @trapexit 

/app/data is a directory under /

/app/data now has alot of data from the application that it is running. If i mount a new disk (sdb) to /app/data, the directory will lost the previous data that are on sda1 partition. I just want to expand the size of /app/data that uses the sda1 partition with the available space of a new disk sdb1 without changing the directory structure and break the application

**Reply by trapexit** [reply]
Yes, but you can simply copy the data over.

**Reply by MichalisDBA** [reply]
Yes that's a solution but for example now the VM has a 1TB disk. We want to add an additional 1TB disk. If i copy the data over we have to add a 2TB disk and we will have 1TB unused that could not take them back because that's how the VM is setuped

**Reply by trapexit** [reply]
You can resize most filesystems. But fine... yes, you can use mergerfs.

**Reply by MichalisDBA** [reply]
So to simplify things. i have data on /dir1 that is mounted on /sda1 partition. If i introduce a new disk sdb and create a partition sdb1, can i use this partition with mergefs to expand the size of /dir1 ?

**Reply by trapexit** [reply, marked as answer]
mergerfs is a union filesystem. It works exactly as described in the docs. It does not work on partitions or raw devices. It works on filesystems. But yes. That's the whole point of a union filesystem. To create a union of other filesystems.

**Reply by MichalisDBA** [reply]
I am running

`mergerfs -o cache.files=partial,dropcacheonclose=true,category.create=mfs /dir1:/dir2 /dir1`

but it says

`* ERROR: branches can not include the mountpoint: /dir1`

Am i doing something wrong? I want to expand the /dir1 with the space available that /dir2 has

**Reply by trapexit** [reply]
As it says you can't recursively include a path. /dir1:/dir2 -> /dir3

**Reply by MichalisDBA** [reply]
Sorry for ignorance, so how do i expand the size of /dir1 ?

**Reply by trapexit** [reply]
Rename dir1 to dir2. merge dir2 and dir3 to dir1.

1) This really isn't anything to do with mergerfs
2) If you're not familiar with Linux or filesystems it would have been useful for you to explain that up front so people can properly offer help.

**Reply by MichalisDBA** [reply]
Thanks for the help. I am familiar but i couldn't wrap my head around how mergefs works because it is new to me. Now i got it and understand what I have to do to accomplish what i want. Thanks for the help and sorry for the hassle again.
